tiledb/sm/query/writer.cc

```diff
@@ -89,7 +89,7 @@ const Array* Writer::array() const {
   return array_;
 }
 
-Status Writer::add_range(unsigned dim_idx, const Range& range) {
+Status Writer::add_range(unsigned dim_idx, Range& range) {
```

Comment:
I understand why the `const` was dropped here, however I think the better approach would be to `std::move` it through the call stack since the callers in the two `Query` routines do need to retain ownership of `range` after calling this function. 

Let's change the arg type to `Range&& range` in this routine, `Reader::add_range`, and `Subarray::add_range`. The `Subarray::add_range` will have ownership and can then mutate/adjust it in isolation from the caller.